src/main/java/gyro/google/compute/InstanceTemplateResource.java

```diff
@@ -0,0 +1,272 @@
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * The instance properties for this instance template.
+     *
+     * @subresource gyro.google.compute.ComputeInstanceProperties
+     */
+    @ConflictsWith("source-instance")
+    public ComputeInstanceProperties getProperties() {
+        return properties;
+    }
+
+    public void setProperties(ComputeInstanceProperties properties) {
+        this.properties = properties;
+    }
+
+    /**
+     * The URL for this instance template. The server defines this URL.
+     */
+    @Output
+    public String getSelfLink() {
+        return selfLink;
+    }
+
+    public void setSelfLink(String selfLink) {
+        this.selfLink = selfLink;
+    }
+
+    /**
+     * The source instance used to create the template.
+     *
+     * @resource gyro.google.compute.InstanceResource
+     */
+    @ConflictsWith("properties")
+    public InstanceResource getSourceInstance() {
+        return sourceInstance;
+    }
+
+    public void setSourceInstance(InstanceResource sourceInstance) {
+        this.sourceInstance = sourceInstance;
+    }
+
+    /**
+     * The source instance params to use to create this instance template.
+     *
+     * @subresource gyro.google.compute.ComputeSourceInstanceParams
+     */
+    @DependsOn("source-instance")
+    public ComputeSourceInstanceParams getSourceInstanceParams() {
+        return sourceInstanceParams;
+    }
+
+    public void setSourceInstanceParams(ComputeSourceInstanceParams sourceInstanceParams) {
+        this.sourceInstanceParams = sourceInstanceParams;
+    }
+
+    @Override
+    protected boolean doRefresh() throws Exception {
+        Compute client = createComputeClient();
+        copyFrom(client.instanceTemplates().get(getProjectId(), getName()).execute());
+        return true;
+    }
+
+    @Override
+    protected void doCreate(GyroUI ui, State state) throws Exception {
+        InstanceTemplate instanceTemplate = new InstanceTemplate();
+        instanceTemplate.setDescription(getDescription());
+        instanceTemplate.setName(getName());
+        ComputeInstanceProperties property = getProperties();
+
+        if (property != null) {
+            instanceTemplate.setProperties(property.toInstanceProperties());
+        }
+        InstanceResource sourceInstance = getSourceInstance();
+
+        if (sourceInstance != null) {
+            instanceTemplate.setSourceInstance(sourceInstance.getSelfLink());
+        }
+        ComputeSourceInstanceParams sourceInstanceParam = getSourceInstanceParams();
+
+        if (sourceInstanceParam != null) {
+            instanceTemplate.setSourceInstanceParams(sourceInstanceParam.toSourceInstanceParams());
+        }
+        Compute client = createComputeClient();
+        waitForCompletion(client, client.instanceTemplates().insert(getProjectId(), instanceTemplate).execute());
+        refresh();
```

Comment:
Line breaks are a bit odd as unrelated statements are clubbed together.

Would be better something like:
```
        instanceTemplate.setName(getName());

        ComputeInstanceProperties property = getProperties();
        if (property != null) {
            instanceTemplate.setProperties(property.toInstanceProperties());
        }

        InstanceResource sourceInstance = getSourceInstance();
        if (sourceInstance != null) {
            instanceTemplate.setSourceInstance(sourceInstance.getSelfLink());
        }

        ComputeSourceInstanceParams sourceInstanceParam = getSourceInstanceParams();
        if (sourceInstanceParam != null) {
            instanceTemplate.setSourceInstanceParams(sourceInstanceParam.toSourceInstanceParams());
        }

        Compute client = createComputeClient();
        waitForCompletion(client, client.instanceTemplates().insert(getProjectId(), instanceTemplate).execute());

        refresh();
```